### PR TITLE
Fix: Nonce

### LINKED ISSUE #5: Bug: Nonce not applied
Reproduce:

1. Create a inline script with with nonce e.g. `<script nonce={{nonce}}>alert(1)</script>` 
2. The nonce is not applied correctly.

---

In the source I can see the `nonce` variable is passed to the view here: https://github.com/GreenMeteor/codebox/blob/master/widgets/CodeboxFrame.php#L34

But on the view side it's not used:
https://github.com/GreenMeteor/codebox/blob/master/widgets/views/codeboxframe.php

### PR DESCRIPTION
**Pull request by ArchBlood**
resolves #5

### CODE REVIEW

File: Module.php [hunk 1]
```diff
@@ -31,12 +32,18 @@ public function getTitle()
...
     {
         $htmlCode = $this->settings->get('htmlCode');
+
         if (empty($htmlCode)) {
             return '';
         }
-        return $htmlCode;
+
+        $nonce = Html::nonce();
+        $htmlNonce = preg_replace('/<script([^>]*)>/', '<script$1 nonce="' . $nonce . '">', $htmlCode);
```
Comment (ArchBlood) [inline Module.php]: If this doesn't work as needed then `$htmlNonce = str_replace('nonce={{nonce}}', $nonce, $htmlCode);` without a value should.

Comment (luke-) [reply Module.php]: Didn't test it but I would expect: 

```suggestion
        $htmlNonce = preg_replace('/<script([^>]*)>/', '<script$1 ' . $nonce . '>', $htmlCode);
```

**Comment by ArchBlood** [comment]
Using either `Security::getNonce()` or `Html::nonce()` seems to only apply `nonce=""` which is a rather strange behavior, in the view file if I were to wrap `$htmlCode` in `Html::` it won't work. So I may have something else to look at later.

**Comment by ArchBlood** [comment]
Okay, so the current method does apply the `nonce="random"` but doesn't display within the widget view, I've checked using `var_dump($nonce)` which does show the correct value, I've also tested other files such as the `Module.php`, `CodeboxFrame.php` and even the `ConfigureForm.php` which all of these do the same thing and return `nonce=""` in the view only, but I can confirm that the nonce is applied correctly in all cases it's just not displaying the value for widget views.

**Comment by ArchBlood** [comment]
Seems like this is an issue within the legal module as well.

**Pull request merged**